public/app/plugins/datasource/tempo/datasource.ts

```diff
@@ -288,14 +288,36 @@ function serviceMapQuery(request: DataQueryRequest<TempoQuery>, datasourceUid: s
         throw new Error(errorRes.error!.message);
       }
 
+      const { nodes, edges } = mapPromMetricsToServiceMap(responses, request.range);
+      nodes.fields[0].config = {
+        links: [
+          makePromLink('Total requests', totalsMetric, datasourceUid),
+          makePromLink('Failed requests', failedMetric, datasourceUid),
+        ],
+      };
+
       return {
-        data: mapPromMetricsToServiceMap(responses, request.range),
+        data: [nodes, edges],
         state: LoadingState.Done,
       };
     })
   );
 }
 
+function makePromLink(title: string, metric: string, datasourceUid: string) {
+  return {
+    url: '',
+    title,
+    internal: {
+      query: {
+        expr: metric,
```

Comment:
Do we want to run here range or instant query or both, cause ideally we should clarify this here.

Reply:
Fixing this in Prometheus in https://github.com/grafana/grafana/pull/41263